room/room-compiler-processing/src/main/java/androidx/room/compiler/processing/XProcessingEnv.kt

```diff
@@ -148,8 +148,7 @@ interface XProcessingEnv {
             options = options,
             codeGenerator = codeGenerator,
             logger = logger,
-            resolver = resolver
-        )
+        ).also { it.resolver = resolver }
```

Comment:
Not for this CL, but I think we'll need to reconsider XProcessingEnv.create() and XRound.create() public APIs as entry points into xprocessing APIs, it seems we've now shifted to XBasicAnnotationProcessor being the entry point since the xprocessing env now requires some management that might be missed by users instantiating them more manually.

Reply:
Yeah, I was thinking about this too. However, I think there's still some benefit to keeping these APIs for now since we need to be able to support processors that cannot use `BasicAnnotationProcessor`. For example, Hilt can't use `BasicAnnotationProcessor` yet due to issues with `SuperficialValidation`.